pkg/kubelet/kubelet.go

```diff
@@ -401,8 +403,22 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 			options.FieldSelector = fields.Set{metav1.ObjectNameField: string(nodeName)}.String()
 		}))
 		nodeLister = kubeInformers.Core().V1().Nodes().Lister()
+		nodeInformer := kubeInformers.Core().V1().Nodes().Informer()
 		nodeHasSynced = func() bool {
-			return kubeInformers.Core().V1().Nodes().Informer().HasSynced()
+			return nodeInformer.HasSynced()
+		}
+		handle, err := nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
I'm curious if we need nodeInformer usage here. What specific signal are we me monitoring for registartion here?

For self registration enabled kubelets, we could check if `registerWithAPIServer` completed successfully? [1]

[1] https://github.com/kubernetes/kubernetes/blob/e9c9a27c97634c433dd9dad1118a98b7e4d519f8/pkg/kubelet/kubelet_node_status.go#L50-L50   

For non self registration kubelets, perhaps we can just check if `GetNode` ever succeded (and did not return the `initialNode`) object [2]

[2] https://github.com/kubernetes/kubernetes/blob/e9c9a27c97634c433dd9dad1118a98b7e4d519f8/pkg/kubelet/kubelet_getters.go#L279-L279

Reply:
The alternate way to do this as you mentioned would be to run a Go routine that periodically polls `GetNode`. When it succeeds, send a message over the channel to sync static pods.

Between the two choices, I felt the node informer would be faster. But maybe the Go routine approach will be cleaner?

Reply:
I think I was the one that suggested this approach.

> Or is the idea that signal of node informer of node object creation will cover both of these cases?

Yes, that's the idea.

Part of the idea of using an informer with an event handler is to minimize time to node ready. Why would a polling approach be preferred?

Reply:
I find the informer approach simpler/easier to understand